diff -ru linux/drivers/media/dvb/frontends/cx24116.c linux/drivers/media/dvb/frontends/cx24116.c
--- linux/drivers/media/dvb/frontends/cx24116.c	2008-11-21 23:00:55.000000000 +0100
+++ linux/drivers/media/dvb/frontends/cx24116.c	2008-11-23 11:36:31.000000000 +0100
@@ -1459,6 +1459,7 @@
 			FE_CAN_FEC_1_2 | FE_CAN_FEC_2_3 | FE_CAN_FEC_3_4 |
 			FE_CAN_FEC_4_5 | FE_CAN_FEC_5_6 | FE_CAN_FEC_6_7 |
 			FE_CAN_FEC_7_8 | FE_CAN_FEC_AUTO |
+			FE_CAN_2ND_GEN_MODULATION |
 			FE_CAN_QPSK | FE_CAN_RECOVER
 	},
 
diff -ru linux/drivers/media/dvb/frontends/stb0899_drv.c linux/drivers/media/dvb/frontends/stb0899_drv.c
--- linux/drivers/media/dvb/frontends/stb0899_drv.c	2008-11-21 23:00:55.000000000 +0100
+++ linux/drivers/media/dvb/frontends/stb0899_drv.c	2008-11-23 11:37:01.000000000 +0100
@@ -1913,6 +1913,7 @@
 
 		.caps 			= FE_CAN_INVERSION_AUTO	|
 					  FE_CAN_FEC_AUTO	|
+					  FE_CAN_2ND_GEN_MODULATION |
 					  FE_CAN_QPSK
 	},
 
diff -ru linux/include/linux/dvb/frontend.h linux/include/linux/dvb/frontend.h
--- linux/include/linux/dvb/frontend.h	2008-11-21 23:00:55.000000000 +0100
+++ linux/include/linux/dvb/frontend.h	2008-11-23 11:27:21.000000000 +0100
@@ -63,6 +63,7 @@
 	FE_CAN_8VSB			= 0x200000,
 	FE_CAN_16VSB			= 0x400000,
 	FE_HAS_EXTENDED_CAPS		= 0x800000,   // We need more bitspace for newer APIs, indicate this.
+        FE_CAN_2ND_GEN_MODULATION       = 0x10000000, // frontend supports "2nd generation modulation" (DVB-S2)
 	FE_NEEDS_BENDING		= 0x20000000, // not supported anymore, don't use (frontend requires frequency bending)
 	FE_CAN_RECOVER			= 0x40000000, // frontend can recover from a cable unplug automatically
 	FE_CAN_MUTE_TS			= 0x80000000  // frontend can stop spurious TS data output
