--- menu.c	2003/05/30 09:53:57	1.253
+++ menu.c	2003/06/06 12:56:05
@@ -615,6 +615,7 @@
            if (channel) {
               *channel = data;
               isyslog("edited channel %d %s", channel->Number(), data.ToText());
+              Timers.Save();
               state = osBack;
               }
            else {
--- svdrp.c	2003/04/27 14:21:07	1.51
+++ svdrp.c	2003/06/06 13:30:52
@@ -790,6 +790,7 @@
                  Channels.ReNumber();
                  Channels.Save();
                  isyslog("modifed channel %d %s", channel->Number(), channel->ToText());
+                 Timers.Save();
                  Reply(250, "%d %s", channel->Number(), channel->ToText());
                  }
               else
