diff -ruN vdr-1.4.3-4/CONTRIBUTORS vdr-1.4.4/CONTRIBUTORS
--- vdr-1.4.3-4/CONTRIBUTORS	2006-10-20 15:42:14.000000000 +0200
+++ vdr-1.4.4/CONTRIBUTORS	2006-11-04 14:18:43.000000000 +0100
@@ -1384,6 +1384,8 @@
  some superfluous semicolons in ci.c
  for pointing out what caused the problem with audio track descriptions after a
  replay has been stopped
+ for reporting a problem with displaying the replay mode symbol in case of "Multi
+ speed mode"
 
 Jürgen Schmitz <j.schmitz@web.de>
  for reporting a bug in displaying the current channel when switching via the SVDRP
diff -ruN vdr-1.4.3-4/HISTORY vdr-1.4.4/HISTORY
--- vdr-1.4.3-4/HISTORY	2006-10-29 11:09:53.000000000 +0100
+++ vdr-1.4.4/HISTORY	2006-11-04 14:18:10.000000000 +0100
@@ -4980,3 +4980,8 @@
 - Fixed deleting EPG events that have a running status of "pausing" or higher.
 - Fixed handling NITs with more than one delivery system descriptor tag for the
   same transponder.
+
+2006-11-04: Version 1.4.4
+
+- Fixed displaying the replay mode symbol in case of "Multi speed mode" (reported
+  by Marco Schlüßler).
diff -ruN vdr-1.4.3-4/config.h vdr-1.4.4/config.h
--- vdr-1.4.3-4/config.h	2006-10-28 11:15:00.000000000 +0200
+++ vdr-1.4.4/config.h	2006-11-04 14:19:31.000000000 +0100
@@ -4,7 +4,7 @@
  * See the main source file 'vdr.c' for copyright information and
  * how to reach the author.
  *
- * $Id: config.h 1.276 2006/10/28 09:15:00 kls Exp $
+ * $Id: config.h 1.277 2006/11/04 13:19:31 kls Exp $
  */
 
 #ifndef __CONFIG_H
@@ -21,8 +21,8 @@
 
 // VDR's own version number:
 
-#define VDRVERSION  "1.4.3-4"
-#define VDRVERSNUM   10403  // Version * 10000 + Major * 100 + Minor
+#define VDRVERSION  "1.4.4"
+#define VDRVERSNUM   10404  // Version * 10000 + Major * 100 + Minor
 
 // The plugin API's version number:
 
diff -ruN vdr-1.4.3-4/menu.c vdr-1.4.4/menu.c
--- vdr-1.4.3-4/menu.c	2006-10-20 15:18:38.000000000 +0200
+++ vdr-1.4.4/menu.c	2006-11-04 14:13:18.000000000 +0100
@@ -4,7 +4,7 @@
  * See the main source file 'vdr.c' for copyright information and
  * how to reach the author.
  *
- * $Id: menu.c 1.444 2006/10/20 13:18:38 kls Exp $
+ * $Id: menu.c 1.445 2006/11/04 13:13:18 kls Exp $
  */
 
 #include "menu.h"
@@ -3818,7 +3818,7 @@
   visible = modeOnly = shown = displayFrames = false;
   lastCurrent = lastTotal = -1;
   lastPlay = lastForward = false;
-  lastSpeed = -1;
+  lastSpeed = -2; // an invalid value
   timeoutShow = 0;
   timeSearchActive = false;
   marks.Load(fileName);
@@ -3885,7 +3885,7 @@
      needsFastResponse = visible = false;
      modeOnly = false;
      lastPlay = lastForward = false;
-     lastSpeed = -1;
+     lastSpeed = -2; // an invalid value
      timeSearchActive = false;
      }
 }
