diff -ruN vdr-1.4.6-1/HISTORY vdr-1.4.7/HISTORY
--- vdr-1.4.6-1/HISTORY	2007-04-30 15:00:51.000000000 +0200
+++ vdr-1.4.7/HISTORY	2007-05-12 11:03:44.000000000 +0200
@@ -5057,3 +5057,7 @@
   (thanks to Marco Schlüßler).
 - Increased the maximum number of CA system ids to cope with the AlphaCrypt
   CAM's version 3.11 firmware.
+
+2007-05-12: Version 1.4.7
+
+- Official release.
diff -ruN vdr-1.4.6-1/config.h vdr-1.4.7/config.h
--- vdr-1.4.6-1/config.h	2007-04-30 11:50:37.000000000 +0200
+++ vdr-1.4.7/config.h	2007-05-12 11:07:16.000000000 +0200
@@ -4,7 +4,7 @@
  * See the main source file 'vdr.c' for copyright information and
  * how to reach the author.
  *
- * $Id: config.h 1.282.1.4 2007/04/30 09:50:37 kls Exp $
+ * $Id: config.h 1.282.1.5 2007/05/12 09:07:16 kls Exp $
  */
 
 #ifndef __CONFIG_H
@@ -21,8 +21,8 @@
 
 // VDR's own version number:
 
-#define VDRVERSION  "1.4.6-1"
-#define VDRVERSNUM   10406  // Version * 10000 + Major * 100 + Minor
+#define VDRVERSION  "1.4.7"
+#define VDRVERSNUM   10407  // Version * 10000 + Major * 100 + Minor
 
 // The plugin API's version number:
 
